File: Neetcode/ValidPalindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = s.lower()
        s = list(s)
        pali = list()
        for char in s:
            if 97 <= ord(char) <= 122 or 48 <= ord(char) <= 57:
                pali.append(char)
        l=0
        r=len(pali)-1
        while l!=r and l<r:
            if pali[l] != pali[r]:
                return False
            l += 1
            r -= 1
        return True

File: Neetcode/ValidPalindrome.cpp
#include <cctype>
class Solution {
public:
    bool isPalindrome(string s) {
        vector<char> sv;
        for(char c: s){
            c = tolower(c);
            if (('a' <= c && c <= 'z') || ('0' <= c && c <= '9')){
                sv.push_back(c);
            }
        }
        int i=0;
        int j=sv.size()-1;
        while (i < j){
            if (sv[i] != sv[j]){
                return false;
            }
            i += 1;
            j -= 1;
        }
        return true;
    }
};
